### PR TITLE
feat: Restrict transfers during lockup period (SC-2045)

### PR DESCRIPTION
**Pull request by michaelelliot**
# Description

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog

### CODE REVIEW

File: contracts/Pool.sol
Comment (lucas-manuel) [inline contracts/Pool.sol:437]: Lets make this multiline now for readability

File: contracts/library/PoolLib.sol
Comment (lucas-manuel) [inline contracts/library/PoolLib.sol:260]: Too much spacing on error messages

File: contracts/test/StakeLocker.t.sol
Comment (lucas-manuel) [inline contracts/test/StakeLocker.t.sol:295]: We should add another fail assert for `hevm.warp(start + stakeLocker.lockupPeriod() - 1);`

File: contracts/test/PoolLiquidityProvider.t.sol
Comment (lucas-manuel) [inline contracts/test/PoolLiquidityProvider.t.sol:199]: We should add another failure assert for `hevm.warp(startDate + pool.lockupPeriod() - 1);`

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Small changes, lets wait to merge this until the interest PR is merged

**Review by lucas-manuel — DISMISSED** [review]
Approving, will merge after https://github.com/maple-labs/maple-core/tree/sc-2046-interest-accrual-bug is in develop

**Review by deluca-mike — DISMISSED** [review]

**Review by deluca-mike — APPROVED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**
